libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -809,6 +809,77 @@ int32_t SOCKETS_SetSockOpt( Socket_t xSocket,
 
             break;
 
+        case SOCKETS_SO_TCPKEEPALIVE:
+           {
+               int keep_alive = *( ( int * ) pvOptionValue );
+
+               if( ( keep_alive == 0 ) || ( keep_alive == 1 ) )
+               {
+                   ret = lwip_setsockopt( ctx->ip_socket,
+                                          SOL_SOCKET,
+                                          SO_KEEPALIVE,
+                                          &keep_alive,
+                                          sizeof( keep_alive ) );
+
+                   if( 0 > ret )
+                   {
+                       return SOCKETS_SOCKET_ERROR;
+                   }
+               }
+               else
+               {
+                   return SOCKETS_EINVAL;
+               }
+
+               break;
+           }
+
+            #if LWIP_TCP_KEEPALIVE
+                case SOCKETS_SO_TCPKEEPALIVE_INTERVAL:
+
+                    ret = lwip_setsockopt( ctx->ip_socket,
+                                           IPPROTO_TCP,
+                                           TCP_KEEPINTVL,
+                                           pvOptionValue,
+                                           sizeof( int ) );
+
+                    if( 0 > ret )
```

Comment:
Lines 846-849 can be moved after line 885 and the blocks at 861-865 and 877-879 can be deleted, _if_ you change the initialization of `ret` [at the top of this function](https://github.com/aws/amazon-freertos/blob/master/libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c#L619):

```c
    int ret = 0;
```

Reply:
I will do this change.